browser/ui/containers/containers_menu_model.cc

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/browser/ui/containers/containers_menu_model.h"
+
+#include <utility>
+#include <vector>
+
+#include "base/strings/utf_string_conversions.h"
+#include "brave/app/brave_command_ids.h"
+#include "brave/components/containers/core/browser/prefs.h"
+#include "brave/grit/brave_generated_resources.h"
+#include "chrome/browser/ui/chrome_pages.h"
+#include "chrome/grit/generated_resources.h"
+
+namespace containers {
+
+namespace {
+
+std::vector<ContainerModel> GetContainerModelsFromPrefs(
+    const PrefService& prefs) {
+  std::vector<ContainerModel> containers;
+  for (auto& container : GetContainersFromPrefs(prefs)) {
+    containers.emplace_back(std::move(container));
+  }
+  return containers;
+}
+
+}  // namespace
+
+ContainersMenuModel::ContainersMenuModel(Delegate& delegate,
+                                         const PrefService& prefs)
+    : ContainersMenuModel(delegate, GetContainerModelsFromPrefs(prefs)) {}
+
+ContainersMenuModel::ContainersMenuModel(Delegate& delegate,
+                                         std::vector<ContainerModel> items)
+    : ui::SimpleMenuModel(this), delegate_(delegate), items_(std::move(items)) {
+  CHECK_LE(items_.size(), static_cast<size_t>(IDC_OPEN_IN_CONTAINER_END -
```

Comment:
let's not crash here, but trim the items list to the maximum possible size and log a warning.

maybe trim the list in `GetContainerModelsFromPrefs`.

Reply:
That's a way better. Thanks